src/controls/widgets/prompt/plugins/Plugin_LoraBoxUI.tsx

```diff
@@ -36,20 +38,40 @@ const LoraBoxUI = observer(function LoraBoxUI_(p: {
     const loraMetadata = st.configFile.value?.loraPrompts?.[loraName]
     const associatedText = loraMetadata?.text ?? ''
     const associatedUrl = loraMetadata?.url ?? ''
+    // const weightedExpression =
+    const weighted = p.uist.ast.findAll('WeightedExpression').find((weighted) => {
+        let text = weighted.contentText
+        if (text.length == 0 || text.charAt(0) != '@') {
+            return false
+        }
+        return weighted.contentText.indexOf(`@${loraName}`) > -1
+    })
+
```

Comment:
not 100% sure I understand what is going on there

Reply:
It might be premature optimization/over-engineering?
I'm checking if the string is actually a lora before using indexOf because otherwise every lora box is going to check every string using indexOf instead of just checking the first character.
It returns the first WeightedExpression that matches this, (This might not be good if someone uses the lora multiple times in the same prompt, but I don't know why someone would do that.)
IndexOf might also just do something like this under the hood though?